hw11/mailroom.py

```diff
@@ -0,0 +1,96 @@
+R - Create a (R)eport\n\nquit - Quit the program\n\n""")
+
+        if userinput == 'T':
+            fulldonorlist = thankyou(userinput, fulldonorlist)
+        elif userinput == 'R':
+            createreport(fulldonorlist)
+        elif userinput != 'quit':
+            print('invalid entry')
+
+
+# Prepare the thank you email
+def thankyou(userinput, fulldonorlist):
+    while userinput != 'quit':
+        userinput = input("""\nPlease enter a name, or choose from the following:\n
+list - Print a list of previous donors\n
+quit - Return to main menu\n\n""")
+        check = False
+        # Initialize the donor list
+        donorlist = fulldonorlist[0]
+        # If the user types list, print the names
+        if userinput == 'list':
+            print(donorlist)
+            userinput = input("Enter Donor Name: ")
+            check = userinput in fulldonorlist[0]
+            if check:
+                ind = fulldonorlist[0].index(userinput)
+            else:
+                fulldonorlist[0].append(userinput)
+        elif userinput != 'quit':
+            # append list of donors with new name
+            # donorlist.append(userinput)
+            check = userinput in fulldonorlist[0]
+            if check:
+                ind = fulldonorlist[0].index(userinput)
+            else:
+                fulldonorlist[0].append(userinput)
+        else:
+            break
+
+        donorname = userinput
+        # Promt for a donation amount
+        donoramount = "none"
+
+        while donoramount == "none":
+            userinput = input("""\nPlease enter a donation \
+amount or 'quit':\n\n""")
+
+            try:
+                donoramount = int(userinput)
+                if check:
+                    fulldonorlist[1][ind].append(donoramount)
+                else:
+                    fulldonorlist[1].append([donoramount])
+                break
+            except ValueError:
+                print("Entry was not a number")
+
+        # Draft the email
+        print('\nDear %s,\n\n' % donorname, 'Thank you so \
+much for your kind donation of $%d.' % donoramount, 'We \
+here at the Foundation for Homeless Whales greatly \
+appreciate it. Your money will go towards creating new \
+oceans on the moon for whales to live in.\n\n\
+Thanks again,\n\nJim Grant\n\nDirector, F.H.W.')
+        input('\nPress Enter to Continue ...')
+        return fulldonorlist
+        break
+
+
+def createreport(fulldonorlist):
+    # read through full donor list
+    totaldonation = []
+    averagedonation = []
+    for i in range(0, len(fulldonorlist[0])):
+        totaldonation.append(sum(int(j) for j in fulldonorlist[1][i]))
+        averagedonation.append(totaldonation[i] / len(fulldonorlist[1][i]))
+        fulldonorlist[1][i] = sum(int(j) for j in fulldonorlist[1][i])
+        fulldonorlist[2].append(averagedonation[i])
+    print(fulldonorlist)
+    sortedlist = fulldonorlist  # somehow magically sort the data
```

Comment:
Here you'll probably want to use sorted or sort on the list. Sorted returns a new copy of the list sorted and sort sorts your list in place. I think the issue you may have encountered is that your donors are not well associated with their donation amounts in the database, such that when you start sorting donation amounts, you lose track of whom they belonged to. Restructuring your donor list might help this. Here's some examples of sorting, https://wiki.python.org/moin/HowTo/Sorting , let me know if you have questions.